model-engine/model_engine_server/api/app.py

```diff
@@ -35,29 +45,31 @@
 app.include_router(file_router_v1)
 app.include_router(trigger_router_v1)
 
-logger = make_logger(filename_wo_ext(__name__))
-
-
-class ExceptionLoggingMiddleware(BaseHTTPMiddleware):
-    async def dispatch(self, request: Request, call_next):
-        try:
-            return await call_next(request)
-        except Exception as e:
-            tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
-            structured_log = {"error": str(e), "traceback": "".join(tb_str)}
-            logger.error("Unhandled exception: %s", structured_log)
-            request_id = get_request_id()
-            return JSONResponse(
-                {
-                    "status_code": 500,
-                    "content": {
-                        "error": f"Internal error for request_id {request_id}. Our team has been notified."
-                    },
-                }
-            )
 
+@app.middleware("http")
+async def dispatch(request: Request, call_next):
+    try:
+        set_request_id(str(uuid.uuid4()))
+        return await call_next(request)
+    except Exception as e:
+        tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
+        request_id = get_request_id()
+        timestamp = datetime.now(pytz.timezone("US/Pacific")).strftime("%Y-%m-%d %H:%M:%S %Z")
+        structured_log = {
+            "error": str(e),
+            "request_id": str(request_id),
+            "traceback": "".join(tb_str),
+        }
+        logger.error("Unhandled exception: %s", structured_log)
+        return JSONResponse(
+            {
+                "status_code": 500,
+                "content": {
+                    "error": f"Internal error for request_id {request_id} at time {timestamp}. Our team has been notified."
```

Comment:
Think I'd rather just have the `error` be "Internal error occurred. Our team has been notified". And then have separate keys for `request_id` and `timestamp`.

Reply:
Also I thought we were going to have a DTO for this?

Reply:
oh yeah I had previously written an InternalError DTO but then didn't ultimately use it, I can go back to using that.

Reply:
[discussed offline] will actually leave it as JSONResponse since handling the new exception thrown by DTO is effectively the same.

Reply:
new format of responses:
```
{"status_code":500,"content":{"error":"Internal error occurred. Our team has been notified.","timestamp":"2023-09-22 16:56:09 PDT","request_id":"60705bc0-8662-45fd-8b15-ecfbcb17cd8f"}}
```